### ISSUE TITLE
socket_path must be a value

### ISSUE THREAD

**Issue by Saran307** [ansible-collections/dellemc.enterprise_sonic#323]
I was trying to execute a simple playbook to shut the interface of sonic nos device but i am getting into error.

currently i am running in ansible 2.14.2
```
ksaran@XXXX:/etc/ansible# ansible --version
ansible [core 2.14.2]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/ksaran/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.11/site-packages/ansible
```
  
This is my playbook 

```
---
- name: shut_test
  hosts: nva1
  become: yes
  gather_facts: no

  tasks:
    - name: shutting the device
      dellemc.enterprise_sonic.sonic_command:
        commands:
          - 'config t'
          - interface Eth1/7
          - shutdown
          - exit
```

and the error messages i am getting is 

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AssertionError: socket_path must be a value
fatal: [XXXXa.net]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python3"}, "changed": false, "module_stderr": "Shared connection to XXXX.net closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/admin/.ansible/tmp/ansible-tmp-1704685988.6898007-1476106-62438890930361/AnsiballZ_sonic_command.py\", line 107, in <module>\r\n    _ansiballz_main()\r\n  File \"/home/admin/.ansible/tmp/ansible-tmp-1704685988.6898007-1476106-62438890930361/AnsiballZ_sonic_command.py\", line 99, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/home/admin/.ansible/tmp/ansible-tmp-1704685988.6898007-1476106-62438890930361/AnsiballZ_sonic_command.py\", line 48, in invoke_module\r\n    run_name='__main__', alter_sys=True)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 205, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 96, in _run_module_code\r\n    mod_name, mod_spec, pkg_name, script_name)\r\n  File \"/usr/lib/python3.7/runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/tmp/ansible_dellemc.enterprise_sonic.sonic_command_payload_ncieeksa/ansible_dellemc.enterprise_sonic.sonic_command_payload.zip/ansible_collections/dellemc/enterprise_sonic/plugins/modules/sonic_command.py\", line 235, in <module>\r\n  File \"/tmp/ansible_dellemc.enterprise_sonic.sonic_command_payload_ncieeksa/ansible_dellemc.enterprise_sonic.sonic_command_payload.zip/ansible_collections/dellemc/enterprise_sonic/plugins/modules/sonic_command.py\", line 207, in main\r\n  File \"/tmp/ansible_dellemc.enterprise_sonic.sonic_command_payload_ncieeksa/ansible_dellemc.enterprise_sonic.sonic_command_payload.zip/ansible_collections/dellemc/enterprise_sonic/plugins/module_utils/network/sonic/sonic.py\", line 121, in run_commands\r\n  File \"/tmp/ansible_dellemc.enterprise_sonic.sonic_command_payload_ncieeksa/ansible_dellemc.enterprise_sonic.sonic_command_payload.zip/ansible_collections/dellemc/enterprise_sonic/plugins/module_utils/network/sonic/sonic.py\", line 52, in get_connection\r\n  File \"/tmp/ansible_dellemc.enterprise_sonic.sonic_command_payload_ncieeksa/ansible_dellemc.enterprise_sonic.sonic_command_payload.zip/ansible_collections/dellemc/enterprise_sonic/plugins/module_utils/network/sonic/sonic.py\", line 66, in get_capabilities\r\n  File \"/tmp/ansible_dellemc.enterprise_sonic.sonic_command_payload_ncieeksa/ansible_dellemc.enterprise_sonic.sonic_command_payload.zip/ansible/module_utils/connection.py\", line 124, in __init__\r\nAssertionError: socket_path must be a value\r\n", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}

**Comment by Saran307** [comment]
Additional thing. i was able to run the play where are running below versions


```
ansible [core 2.13.10]
  config file = /root/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, May 26 2023, 14:05:08) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True
```
```
root@XXXX:~# python3 --version
Python 3.8.10
```

**Comment by dataolle** [comment]
i had a similar problem and i fixed it by setting the following variables for the switch:
```
ansible_network_os=dellemc.enterprise_sonic.sonic
ansible_httpapi_use_ssl=true
ansible_httpapi_validate_certs=false
```

**Comment by Saran307** [comment]
I checked after adding it but it did not work. Still getting the same error. I am not sure why its working in a different version. Should i have to changes my versions or should i change my path of installation of ansible?

**Comment by kerry-meyer** [comment]
If you don't already have the following line in your host inventory file, please add it.

ansible_connection=ansible.netcommon.network_cli

If that doesn't resolve the problem, please provide the following information:

output from:
- ansible-galaxy collection list
- pip show paramiko

**Comment by kerry-meyer** [comment]
I am adding a link to a related, generic Ansible issue. This reference is for illustration of the fact that this issue is not associated specifically with the Dell enterprise_sonic resource module collection, but is a common consideration for working with the generic Ansible infrastucture.

REF: [ "socket_path must be a value" with get_facts #73363 ](https://github.com/ansible/ansible/issues/73363)

**Comment by Saran307** [comment]
Hey Kerry, Thanks for that help, I did verified the pip show paramiko. It looks like ansible was not installed using pip which was the reason for it. I reinstalled ansible using pip and then i installed paramiko using pip which sorted out the problem.

**Comment by kerry-meyer** [comment]
I'm glad to hear that you were able to fix the problem with a re-installation. 

In addition to the solutions I described, it is also possible that the ansible.cfg file you are using (in "~/.ansible.cfg", "/etc/ansible/ansible.cfg", or some other possible locations) is specifying the correct default connection type for this playbook. If there was some problem with the previous installation, this setting may not have been correctly incorporated as an environment variable.

Since the problem has been resolved, I am changing the status of this issue to "Closed".
